internal/models/permission_record.go

```diff
@@ -4,6 +4,19 @@ import (
 	"time"
 )
 
+// TemplateData contains the core permission data
+type TemplateData struct {
+	Owner      Address             `json:"owner"`
+	Agreements []TemplateAgreement `json:"agreements"`
+}
+
+// TemplateAgreement represents a permission agreement for an asset
+type TemplateAgreement struct {
+	Type        string       `json:"type"`
+	Asset       string       `json:"asset"`
```

Comment:
I don't think templates should have asset in them then that binds them to 1 template per vehicle. And the original goal was a manufacture can have a template used by all vehicles.
```suggestion
	Asset       string       `json:"asset"`
```